File: src/jav/l1_variables/Nullable_type.java
package jav.l1_variables;

public class Nullable_type {
    private static String name = null;
    private static String name2;

    public static void main(String[] args) {
        //System.out.println(name.length());//        Exception in thread "main" java.lang.NullPointerException at jav.Nullable_type.main(Nullable_type.java:6)
       // System.out.println(name2.length());//       Exception in thread "main" java.lang.NullPointerException at jav.Nullable_type.main(Nullable_type.java:6)


        // эквивалент ? в KOTLIN
        if(name!=null){
            System.out.println(name.length());
        } else
        {
            System.out.println("null");
        }

    }
}

File: src/kotl/l1_variables/Nullable_type.kt
package kotl.l1_variables

//private var name: String = null  // Error:(3, 27) Kotlin: Null can not be a value of a non-null type String
private var name: String? = null

fun main() {
     // рператор Элвис
    val length= name?.length?:0
    println(length)

   /* Ситуация с NPE в kotlin*/
    var b: String? = null
    val l = b!!.length
    println(l)
    /*Exception in thread "main" kotlin.KotlinNullPointerException
	at kotl.l1_variables.Nullable_typeKt.main(Nullable_type.kt:11)
	at kotl.l1_variables.Nullable_typeKt.main(Nullable_type.kt)
    * */



}
